planner: update doc for Instance plan cache (#20155)
pingcap/docs: sql-prepared-plan-cache.md

1 row in set (0.00 sec)
```

## Instance Plan Cache

> **Warning:**
>

> > Instance Plan Cache only supported after v8.4 and it is an experimental feature.

Reviewer: ```suggestion
> Instance Plan Cache is experimental feature.
```

> TiDB supports both Session Plan Cache and Instance Plan Cache. Session Plan Cache maintains cache for each session, plans can't be shared across different sessions, while Instance Plan Cache maintains one single cache in the instance for all sessions.

Reviewer: ```suggestion
TiDB supports both Session Plan Cache and Instance Plan Cache. Session Plan Cache maintains cache for each session, while Instance Plan Cache maintains one single cache in each instance which can be shared across all sessions in this instance.
```

Setting `tidb_enable_instance_plan_cache` to true to enable Instance Plan Cache and plans can be shared across different sessions with this feature:

```sql
-- execute below SQLs in session-1
mysql> set global tidb_enable_instance_plan_cache=1;
Query OK, 0 rows affected (0.01 sec)

mysql> prepare st from "select a from t";
Query OK, 0 rows affected (0.01 sec)

mysql> execute st;
Empty set (0.00 sec)

-- execute below SQLs in session-2
mysql> prepare st from "select a from t";
Query OK, 0 rows affected (0.00 sec)

mysql> execute st;
Empty set (0.00 sec)

mysql> select @@last_plan_from_cache;
+------------------------+
| @@last_plan_from_cache |
+------------------------+
|                      1 |
+------------------------+
```

> And for Instance Plan Cache, TiDB provides 2 system views (after v8.5) `information_schema.tidb_plan_cache` and `information_schema.cluster_tidb_plan_cache` to see the internal information of the Instance Plan Cache of the current instance and the whole cluster: 

Reviewer: ```suggestion
In v8.5.0 and higher, two system tables were introduced to observe the entries in Instance Plan Cache: 
* `information_schema.tidb_plan_cache`: The entries of instance plan cache in this particular TiDB instance.
* `information_schema.cluster_tidb_plan_cache`:  The entries of instance plan cache from all available TiDB instances. 
```

1 row in set (0.00 sec)
```

## Instance Plan Cache

> **Warning:**
>

> > Instance Plan Cache is experimental feature.

Reviewer 2: ```suggestion
> Instance Plan Cache is an experimental feature. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
```

> TiDB supports both Session Plan Cache and Instance Plan Cache. Session Plan Cache maintains cache for each session, while Instance Plan Cache maintains one single cache in each instance which can be shared across all sessions in this instance.

Reviewer 2: ```suggestion
TiDB supports both session-level plan cache and instance-level plan cache. Session Plan Cache maintains a separate cache for each session, while Instance Plan Cache maintains a single shared cache across all sessions in an instance.
```

> Setting `tidb_enable_instance_plan_cache` to true to enable Instance Plan Cache and plans can be shared across different sessions with this feature:

Reviewer 2: ```suggestion
To enable Instance Plan Cache, setting [`tidb_enable_instance_plan_cache`](/system-variables.md#tidb_enable_instance_plan_cache-new-in-v840) to `ON`. This allows plans to be shared across different sessions:
```

```sql

> -- execute below SQLs in session-1
> mysql> set global tidb_enable_instance_plan_cache=1;
> Query OK, 0 rows affected (0.01 sec)
> 
> mysql> prepare st from "select a from t";
> Query OK, 0 rows affected (0.01 sec)
> 
> mysql> execute st;
> Empty set (0.00 sec)
> 
> -- execute below SQLs in session-2
> mysql> prepare st from "select a from t";
> Query OK, 0 rows affected (0.00 sec)
> 
> mysql> execute st;
> Empty set (0.00 sec)
> 
> mysql> select @@last_plan_from_cache;
> +------------------------+
> | @@last_plan_from_cache |
> +------------------------+
> |                      1 |
> +------------------------+

Reviewer 2: ```suggestion
-- Execute the following SQL statements in session 1.
mysql> SET GLOBAL tidb_enable_instance_plan_cache=1;
Query OK, 0 rows affected (0.01 sec)

mysql> PREPARE st FROM "SELECT a FROM t";
Query OK, 0 rows affected (0.01 sec)

mysql> EXECUTE st;
Empty set (0.00 sec)

-- Execute the following SQL statements in session 2.
mysql> PREPARE st FROM "SELECT a FROM t";
Query OK, 0 rows affected (0.00 sec)

mysql> EXECUTE st;
Empty set (0.00 sec)

mysql> SELECT @@last_plan_from_cache;
+------------------------+
| @@last_plan_from_cache |
+------------------------+
|                      1 |
+------------------------+
```

```

> 
> In v8.5.0 and higher, two system tables were introduced to observe the entries in Instance Plan Cache: 

Reviewer 2: ```suggestion
In v8.5.0, TiDB introduces two system tables to help you view entries in Instance Plan Cache:

```

> * `information_schema.tidb_plan_cache`: The entries of instance plan cache in this particular TiDB instance.
> * `information_schema.cluster_tidb_plan_cache`:  The entries of instance plan cache from all available TiDB instances. 

Reviewer 2: ```suggestion
* `information_schema.tidb_plan_cache`: shows entries of instance plan cache in the current TiDB instance.
* `information_schema.cluster_tidb_plan_cache`: shows entries of instance plan cache from all available TiDB instances.
```

                             BINDING: 
                             OPT_ENV: f20c20a72b2a33c5c44e805dbea0fa97028e6f047320928cf367f74c8c94737b
                        PARSE_VALUES: 1
                            MEM_SIZE: 13322
                          EXECUTIONS: 1
                      PROCESSED_KEYS: 0
                          TOTAL_KEYS: 0
                         SUM_LATENCY: 5919417
                           LOAD_TIME: 2024-12-05 15:41:43
                    LAST_ACTIVE_TIME: 2024-12-05 15:41:43
tidb_decode_binary_plan(binary_plan): 
| id                  | estRows  | estCost    | task      | access object | operator info                   |
| TableReader_7       | 3323.33  | 372904.43  | root      |               | data:Selection_6                |
| └─Selection_6       | 3323.33  | 5383000.00 | cop[tikv] |               | lt(test.t.a, 1)                 |
|   └─TableFullScan_5 | 10000.00 | 4884000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo  |

> 

Reviewer 2: ```suggestion
```
